pkg/k8s/client.go

```diff
@@ -105,11 +108,50 @@ func CreateConfig() (*rest.Config, error) {
 	return createConfig(GetAPIServerURL(), GetKubeconfigPath(), GetQPS(), GetBurst())
 }
 
+func setDialer(config *rest.Config) func() {
+	context := (&net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+	}).DialContext
+	dialer := connrotation.NewDialer(context)
+	config.Dial = dialer.DialContext
+	return dialer.CloseAll
+}
+
+func runHeartbeat(closeAllConns []func(), stop chan struct{}) error {
+	timeout := option.Config.K8sHeartbeatTimeout
+	go wait.Until(func() {
+		done := make(chan error)
+		go func() {
+			_, err := k8sCli.Interface.CoreV1().Namespaces().Get("kube-system", metav1.GetOptions{})
```

Comment:
please add this comment to the code:
```
Kubernetes does a get node of the node that kubelet is running [0]. This seems excessive in our case because the amount of data transferred is bigger than doing a Get of the kube-system namespace. For this reason we have picked to perform a get on `kube-system` instead a get of a node.

[0] https://github.com/kubernetes/kubernetes/blob/v1.17.3/pkg/kubelet/kubelet_node_status.go#L423
```